routes/auth.router.js

```diff
@@ -0,0 +1,70 @@
+const express = require('express');
+const mongoose = require('mongoose');
+const {generateJWT} = require('../jwt');
+const bcrypt = require('bcryptjs');
+const router = express.Router();
+const bcryptSalt = bcrypt.genSaltSync(8);
+
+const User = require('../models/user.model');
+const Cart = require('../models/cart.model');
+const Wishlist = require('../models/wishlist.model');
+
+router.post('/signup', async function(req, res) {
+  const { email, password, name } = req.body;
+  const existingUser = await checkIfUserExists(email);
+  if (!existingUser) {
+    const hashedPassword = bcrypt.hashSync(password, bcryptSalt);   
+    const user = await User.create({ name: name, email: email, password: hashedPassword });
+    const cart = await Cart.create({uid: user._id});
+    const wishlist = await Wishlist.create({uid: user._id});
+    const token = generateJWT(user._id.toString());
+
+    res.json({
+      success: true,
+      token: token
+    });
```

Comment:
res.json({ success:true, token }) would also work

Reply:
yes, keeping the key value to pair to ease the readability